internal/metrics/remote_write_test.go

```diff
@@ -25,21 +26,70 @@ var _ = Describe("remote write", func() {
 			tsdbInstance metrics.API
+			caStr        = `-----BEGIN CERTIFICATE-----
+MIIFiDCCA3CgAwIBAgIUWlVM1UdWMTANbe+XsXplZsvVrHQwDQYJKoZIhvcNAQEL
+BQAwSTELMAkGA1UEBhMCRVUxCzAJBgNVBAgMAk5vMQ4wDAYDVQQHDAVTdGF0ZTEK
+MAgGA1UECgwBRDERMA8GA1UEAwwIcmVjZWl2ZXIwHhcNMjIwNDA0MDgzMjUxWhcN
+MzIwNDAxMDgzMjUxWjBJMQswCQYDVQQGEwJFVTELMAkGA1UECAwCTm8xDjAMBgNV
+BAcMBVN0YXRlMQowCAYDVQQKDAFEMREwDwYDVQQDDAhyZWNlaXZlcjCCAiIwDQYJ
+KoZIhvcNAQEBBQADggIPADCCAgoCggIBALyt9dKvRF2BP5nRjcDzDL7erP0X1dzp
+gzqX5j0O+CsrI8Iw/BngWF4BQ8LBgHkI8vzXv3H80BCPTDo7D+GEkQBM3oYINp+v
+VjOJ2WsY9KJ5tG/Qdk40QHK8PluCIIvfZd1ZuCksNQaeSoBBhO2uDimjRuQr2j4j
+rA7+tDl5ijo+Owk6cXxZSa6YV5LhMV55+U7hQq6BIMxFsfuyOcfeG0fqiMdlrlnC
+riEm0EPGkGjOW8tdXgtNpg3DhOFNcMtk+sgGL8BNydRUVJh6g7lvmgll/HL6usAU
+pjb0i+B7bNoHznFhw71G4QtaHO/bGnRer7eOv7/JyEg456ILhy3SOvDRMxueKN3g
+6qyN2lb405AfElEzr9NBTQQihhItVaDFa2JfP02g59MqsTC+BgNyKk25lPzPrAHo
+6Lgp9m7nmWWvj/8kze3VLNVhqmAM8Bhh+50PEmpYOP5K/w1PLdI/VRVCXxinu/Z4
+XzxU3do5g8kjMqQJwkHXXV5zsLsBYHJP5/P2cvuGlYbWx8pLdMti36lGx8qtHhXS
+x/41SIySjc6YxEOt3DSGG0ZcU3Fn+KRnZm8Pl1lQwuioSzOimKlAkn1yIjKgqdMI
+beOuiu6D00inYB3ZEEuQVXtCiCHQNligR/k2Qej2cdsedazFDGVjjSBjWhGKDmXo
+4p3AoiJSHv6XAgMBAAGjaDBmMB0GA1UdDgQWBBQsOQ3FVc7Ko6fpN/UuVwTTcsX2
+BTAfBgNVHSMEGDAWgBQsOQ3FVc7Ko6fpN/UuVwTTcsX2BTAPBgNVHRMBAf8EBTAD
+AQH/MBMGA1UdEQQMMAqCCHJlY2VpdmVyMA0GCSqGSIb3DQEBCwUAA4ICAQCIweEe
+KROsR1jLJFT3s+hQzI0TCCdKsbCHRhMAFttRhmE42gTyzqggJdpwuNA0wfDLyQcg
+KGHCqG22KtyoBUoTiHuQxh7gkMv6rxCHgEbwKjF2XCUm3KZ1lH5I7LtK5VzzCRFu
+xgGA2mvK7nG4KMIUOChSJiBLAti6MP5Azw861qAXSARSd2721sdUzAScrZa0KpBj
+WVqpxg9Hlb4Ad4tApoj1q8MTOdClESSTJJSynQ796Amg3Indgj2qDyQ70SaJd8s9
+tNujtWro+66Wlt2jE2S8S1yL2tRb1tNp43Yqg8IlqGUU/fLVjiNI6KQ+onGhOkOt
+nTboLb4xa2C6dgNYi80NTnZ3mETnkYC/sLaA3yv2HB9zfG+MhT69H8zRsSDmzHRX
+oPWv326JhQwtrBgwXifd8OI43LPIeGNxM+ElyNJZOeQVySb9TMpqk9gTsfLDb7fT
+8h2vlPnQMNiUfn1B4hJ0Sm5CMlZ28ig1MF1YDy1DUWlnzeGOcBgz4c1l0J1wwHto
+XEC7dPo8bMY7D8RCD7e9uKm9MbFntQSt+yDR2RftyxNfoPv42C6d35f4xNOJJv0E
+9p16bl/jBfi66oVfZqypLMu7Xm4ChJz/pEY4HmiMVZ0Xo/BIlUG4NDxve2qU0CA9
+W5WPOpTBuB5eJ1Clr/7QePRsMcRMOCTRloMTQg==
+-----END CERTIFICATE-----`
 		)
 
+		listCaFiles := func() (result []string) {
+			files, err := ioutil.ReadDir(tmpDir)
+			Expect(err).To(BeNil())
+			for _, file := range files {
+				match, err := path.Match(metrics.ServerCAFileNamePattern, file.Name())
+				Expect(err).To(BeNil())
+				if match {
+					result = append(result, file.Name())
+				}
+			}
+			return
+		}
+
 		BeforeEach(func() {
-			tmpDir, err := ioutil.TempDir("", "metrics")
+			var err error
+			tmpDir, err = ioutil.TempDir("", "metrics")
 			Expect(err).ToNot(HaveOccurred())
 
 			tsdbInstance, err = metrics.NewTSDB(tmpDir)
 			Expect(err).NotTo(HaveOccurred())
 
-			remoteWrite = metrics.NewRemoteWrite("", deviceID, tsdbInstance)
+			remoteWrite = metrics.NewRemoteWrite(tmpDir, deviceID, tsdbInstance)
 		})
 
 		AfterEach(func() {
 			err := tsdbInstance.Deregister()
+			Expect(tmpDir).NotTo(BeEmpty())
+			os.RemoveAll(tmpDir)
 			Expect(err).ToNot(HaveOccurred())
```

Comment:
`err` in line 93 refers to line 90.
Please move 
```
   Expect(tmpDir).NotTo(BeEmpty())
   os.RemoveAll(tmpDir)
```
below it

Reply:
done